openpype/hosts/blender/plugins/publish/extract_fbx_animation.py

```diff
@@ -19,20 +22,16 @@ class ExtractAnimationFBX(publish.Extractor):
     optional = True
 
     def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
         # Define extract output file path
         stagingdir = self.staging_dir(instance)
 
         # Perform extraction
         self.log.debug("Performing extraction..")
 
-        # The first collection object in the instance is taken, as there
-        # should be only one that contains the asset group.
-        collection = [
-            obj for obj in instance if type(obj) is bpy.types.Collection][0]
-
-        # Again, the first object in the collection is taken , as there
-        # should be only the asset group in the collection.
-        asset_group = collection.objects[0]
+        asset_group = instance.data["transientData"]["instance_node"]
 
         armature = [
             obj for obj in asset_group.children if obj.type == 'ARMATURE'][0]
```

Comment:
This should be something like this:

```
        collection = instance.data["transientData"]["instance_node"]
        asset_group = collection.objects[0]

        armature = [
            obj for obj in asset_group.children if obj.type == 'ARMATURE'][0]
```

Because `asset_group` is a collection, and `children` returns all the children collections.
I've tried with `asset_group.all_objects`, but it doesn't look to be working because it returns only the objects that are linked in the collections. I don't remember if this has always been the case, I think I remember that it once returned the children of linked objects even if those weren't linked, but maybe I'm not remember this correctly! In any case, it doesn't work, so it needs to be changed :)

Reply:
I suspect this should do it: https://github.com/ynput/OpenPype/pull/5910/commits/d148671d02563ce11413e7cb4fd010f9fb3e41f1

It seems that:

- `Collection.objects`: Children `bpy.types.Object` of this collection, excluding child collections
- `Collection.all_objects`: All children `bpy.types.Object` including children of child collections but excluding all `bpy.types.Collection`
- `Collection.children`: Children `bpy.types.Collection` of this collection, excluding child objects
- `Collection.children_recursive`: All children`bpy.types.Collections` of this and including children of child collections recursively.

As such, none of these find the children (parenting hierarchy) of the `bpy.types.Object`. They do not traverse down the `bpy.types.Object` hierarchy. It requires a call to `Object.children` or `Object.children_recursive` to do so.

For now with this commit I opted to for all children objects of the instance collection (not looking into child collections) search the ARMATURE among all its children hierarchy of `bpy.types.Object`.

Reply:
In this case a `next` would be better. That way instead of looping through all child objects THEN choosing the first one, it stops at the first matching object found.

Reply:
Correct, but in this case I want to log if there are multiple so that's why I continue the full loop. Just so an edge case will be found easier.